packages/fx-core/src/component/coordinator/index.ts

```diff
@@ -290,6 +291,69 @@ export class Coordinator {
     return ok(undefined);
   }
 
+  async preProvisionForVS(
+    ctx: DriverContext,
+    inputs: InputsWithProjectPath
+  ): Promise<
+    Result<
+      {
+        needAzureLogin: boolean;
+        needM365Login: boolean;
+        resolvedAzureSubscriptionId?: string;
+        resolvedAzureResourceGroupName?: string;
+      },
+      FxError
+    >
+  > {
+    const res: {
+      needAzureLogin: boolean;
+      needM365Login: boolean;
+      resolvedAzureSubscriptionId?: string;
+      resolvedAzureResourceGroupName?: string;
+    } = {
+      needAzureLogin: false,
+      needM365Login: false,
+    };
+
+    // 1. parse yml to cycles
+    const parser = new YamlParser();
+    const templatePath =
+      inputs["workflowFilePath"] ??
+      path.join(ctx.projectPath, SettingsFolderName, workflowFileName);
+    const maybeProjectModel = await parser.parse(templatePath);
+    if (maybeProjectModel.isErr()) {
+      return err(maybeProjectModel.error);
+    }
+    const projectModel = maybeProjectModel.value;
+    const cycles: ILifecycle[] = [
+      projectModel.registerApp,
+      projectModel.provision,
+      projectModel.configureApp,
+    ].filter((c) => c !== undefined) as ILifecycle[];
+
+    // 2. check each cycle
+    for (const cycle of cycles) {
+      cycle.resolvePlaceholders();
```

Comment:
Maybe it's better to check the return value of `resolvePlaceholders()` to get unresolved placeholders,

Reply:
There is no need to check whether placeholders are resolved or not, we only need the final value